proto/prysm/v1alpha1/attestation/id.go

```diff
@@ -22,12 +21,12 @@ const (
 )
 
 // Id represents an attestation ID. Its uniqueness depends on the IdSource provided when constructing the Id.
-type Id [33]byte
```

Comment:
Why change this?

Reply:
This is what Potuz told me:

> I will strongly recommend that you just replkace this with [32]bytes and do the first byte to be the version, and the remaining 31 to be the hash[1:32]

> having slices of unaligned data will either double the memory footprint of all these instances or add CPU latency to them

> unnecessarily since you don't really care about the full hash anyway it seems